lib/Parse/Parser.cpp

```diff
@@ -1015,9 +1015,19 @@ Parser::parseList(tok RightK, SourceLoc LeftLoc, SourceLoc &RightLoc,
       }
       break;
     }
+
+    // If the next token is at the beginning of a new line, could be an argument
+    // label, is followed by a colon, and separator omission is permitted, 
+    // expect that we'll parse another list element which begins with that 
+    // argument label and continue parsing the list.
+    if (Tok.isAtStartOfLine() && AllowSepOmission && Tok.canBeArgumentLabel() 
+        && peekToken().is(tok::colon)) { 
+      continue;
```

Comment:
Optional: It would be great to unify that into a single block `if (Tok.isAtStartOfLine()) { if (allowed && cond) continue; if (invalid) break; }` if possible.

Reply:
Changed to

```
    if (Tok.isAtStartOfLine()) {
      if (AllowSepOmission && Tok.canBeArgumentLabel()
          && peekToken().is(tok::colon))
        // If the next token is at the beginning of a new line, could be an
        // argument label, is followed by a colon, and separator omission is
        // permitted, expect that we'll parse another list element which begins
        // with that argument label and continue parsing the list.
        continue;
      else if (Tok.is(tok::r_brace) || isStartOfDecl() || isStartOfStmt())
        // If we're in a comma-separated list, the next token is at the
        // beginning of a new line and can never start an element, break.
        break;
    }
``` 

in https://github.com/apple/swift/pull/22714/commits/4b8c2f9b06e64ab2415febdb840f2781220ec707 . Is this what you have in mind?

Reply:
Yes, thank you!